test/unit/specs/store/search-store.spec.js

```diff
@@ -240,238 +297,144 @@ describe('Search Store', () => {
-      })
-      done()
+      expect(commit).not.toHaveBeenCalledWith(RESET_MEDIA, { mediaType })
     })
 
-    it('FETCH_MEDIA resets images if page is not defined', (done) => {
-      const params = {
-        q: 'foo',
-        page: undefined,
-        shouldPersistMedia: false,
-        mediaType: IMAGE,
-      }
-      const action = store.actions(audioServiceMock, imageServiceMock)[
-        FETCH_MEDIA
-      ]
-      action({ commit, dispatch, state, rootState }, params).then(() => {
-        expect(commit).toBeCalledWith(FETCH_START_MEDIA, {
-          mediaType: IMAGE,
-        })
+    it('FETCH_AUDIO on success', async () => {
+      const mediaType = AUDIO
+      const params = { id: 'foo' }
+      const action = store.actions(services)[FETCH_AUDIO]
+      await action({ commit, dispatch, state, rootState }, params)
+      expect(commit).toHaveBeenCalledWith(FETCH_START_MEDIA, { mediaType })
+      expect(commit).toHaveBeenCalledWith(SET_AUDIO, { audio: {} })
+      expect(commit).toHaveBeenCalledWith(FETCH_END_MEDIA, { mediaType })
+      expect(commit).toHaveBeenCalledWith(SET_AUDIO, { audio: audioDetailData })
+      expect(audioServiceMock.getMediaDetail).toHaveBeenCalledWith(params)
+    })
 
-        expect(commit).toBeCalledWith(FETCH_END_MEDIA, {
-          mediaType: IMAGE,
-        })
+    it('FETCH_AUDIO dispatches SEND_RESULT_CLICKED_EVENT', () => {
+      const params = { id: 'foo' }
+      const action = store.actions(services)[FETCH_AUDIO]
+      action({ commit, dispatch, state, rootState }, params)
 
-        expect(commit).toBeCalledWith(SET_MEDIA, {
-          media: [],
-          mediaType: IMAGE,
-        })
-        done()
-      })
+      expect(dispatch).toHaveBeenLastCalledWith(
+        `${USAGE_DATA}/${SEND_RESULT_CLICKED_EVENT}`,
+        {
+          query: state.query.q,
+          resultUuid: 'foo',
+          resultRank: 0,
+          sessionId: rootState.user.usageSessionId,
+        }
+      )
     })
 
-    it('FETCH_MEDIA does not reset images if page is defined', (done) => {
-      const params = {
-        q: 'foo',
-        page: 1,
-        shouldPersistMedia: false,
-        mediaType: IMAGE,
+    it('FETCH_AUDIO on error', (done) => {
+      services[AUDIO] = {
+        getMediaDetail: jest.fn(() => Promise.reject('error')),
       }
-      const action = store.actions(audioServiceMock, imageServiceMock)[
-        FETCH_MEDIA
-      ]
-      action({ commit, dispatch, state }, params).then(() => {
-        expect(commit).not.toBeCalledWith(SET_MEDIA, { media: [] })
-        done()
+      const params = { id: 'foo' }
+      const action = store.actions(services)[FETCH_AUDIO]
+      action({ commit, dispatch, state, rootState }, params).catch((error) => {
+        expect(commit).toBeCalledWith(FETCH_START_MEDIA, { mediaType: AUDIO })
+        expect(dispatch).toBeCalledWith('HANDLE_MEDIA_ERROR', error)
       })
+      done()
```

Comment:
Ditto here. We can remove the `done` usage completely if we use `expect().rejects`